src/wic/api/pythonapi.py

```diff
@@ -0,0 +1,364 @@
+        else:
+            return super().__setattr__(__name, __value)
+
+    def __getattribute__(self, __name: str) -> Any:
+        if __name != "_input_names" and hasattr(self, "_input_names"):
+            if __name in self._input_names:
+                index = self._input_names.index(__name)
+                return self.inputs[index]
+        return super().__getattribute__(__name)
+
+    def _set_from_io_cfg(self) -> None:
+        for name, value in self.cfg_yaml.items():
+            value_ = _get_value_from_cfg(value)
+            setattr(self, name, value_)
+
+    def _validate(self) -> None:
+        for inp in self.inputs:
+            if inp.required and inp.value is None:
+                raise MissingRequiredValue(f"{inp.name} is required")
+
+    @property
+    def _yml(self) -> dict:
+        d = {
+            self.cwl_name: {
+                "in": {
+                    inp.name: _value_str(inp.value)
+                    for inp in self.inputs
+                    if inp.value is not None
+                }
+            }
+        }
+        return d
+
+    def _save_cwl(self, path: Path) -> None:
+        cwl_adapters = path.joinpath("cwl_adapters")
+        cwl_adapters.mkdir(exist_ok=True)
+        with open(
+            cwl_adapters.joinpath(f"{self.cwl_name}.cwl"),
+            "w",
+            encoding="utf-8",
+        ) as file:
+            file.write(yaml.dump(self.yaml))
+
+
+@dataclass
+class Workflow:
+    steps: list
+    name: str
+    path: Path
+    yml_path: Path = field(init=False)
+
+    def step(self, step_: Step) -> None:
+        """Append step to Workflow."""
+        self.steps.append(step_)
+
+    def __post_init__(self) -> None:
+        for s in self.steps:
+            try:
+                s._validate()  # pylint: disable=W0212
+            except BaseException as exc:
+                raise InvalidStep(f"{s.cwl_name} is missing required inputs") from exc
+
+    @property
+    def yaml(self) -> Any:
+        """WIC YML representation."""
+        d = {"steps": [step._yml for step in self.steps]}
+        return d
+
+    def _save_yaml(self) -> None:
+        self.yml_path = self.path.joinpath(f"{self.name}.yml")
+        with open(self.yml_path, "w", encoding="utf-8") as file:
+            file.write(yaml.dump(self.yaml))
+
+    def _save_all_cwl(self) -> None:
+        for s in self.steps:
+            try:
+                s._save_cwl(self.path)
+            except BaseException as exc:
+                raise exc
+
+    def compile(self) -> Path:
+        """Compile Workflow using WIC."""
+
+        self._save_all_cwl()
+        self._save_yaml()
+        print(f"Compiling {self.name}")
+        proc = subprocess.run(
+            args=["wic", "--yaml", self.yml_path],
+            capture_output=True,
+            cwd=self.path,
+            check=True,
+            text=True,
+            universal_newlines=True,
+        )
+        print(proc.stdout)
```

Comment:
Replace "print" with by using logger.info() as you already using logging package

Reply:
Addressed in #135